src/app/app.component.ts

```diff
@@ -9,5 +11,17 @@ import { MatDialog } from '@angular/material/dialog';
 export class AppComponent implements OnInit {
   constructor(public form: MatDialog) {}
 
-  ngOnInit() {}
+  ngOnInit() {
+    if (localStorage.getItem('token') === null) {
+      this.openAuthForm();
+    }
+  }
+
+  openAuthForm() {
+    let formInfo = this.form.open(AuthorizationComponent);
+  }
+
+  isAdmin() {
+    return localStorage.getItem('role') === 'Admin';
```

Comment:
better to create enum for roles and use like `localStorage.getItem('role') === myEnum.Admin`, less probability to have typo